Index: src/Plugin/facets/facet_source/SearchApiViewsDeriver.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- src/Plugin/facets/facet_source/SearchApiViewsDeriver.php	(date 1481147042000)
+++ src/Plugin/facets/facet_source/SearchApiViewsDeriver.php	(revision )
@@ -44,7 +44,7 @@
             // We only support pages and blocks because those are the ones that
             // we've tested. They are also the only ones that support for
             // ::isRenderedInCurrentRequest() and ::getPath().
-            if (in_array($display_info['display_plugin'], ['page', 'block'])) {
+            if (in_array($display_info['display_plugin'], ['page', 'block', 'default'])) {
               $machine_name = $view->id() . PluginBase::DERIVATIVE_SEPARATOR . $display_id;
 
               $label_arguments = [
